lib/shaman/deploy.rb

```diff
@@ -1,10 +1,18 @@
+require 'yaml'
+
 module Shaman
   class Deploy
     include Helpers
 
+    DEFAULT_ENVIRONMENT = 'default'.freeze
+    DEFAULT_EDITOR = 'vi'.freeze
+    DEFAULT_MESSAGE = ''.freeze
+    DEFAULT_RELEASE_NAME = ''.freeze
+    GIT_DEFAULT_DIRECTORY = '.'.freeze
+    GIT_HEAD = 'HEAD'.freeze
+
     def initialize(args, options)
-      error!('Must specify environment') if args.count < 1
-      @environment = args.first
+      @environment = args.first || DEFAULT_ENVIRONMENT
```

Comment:
Do all past, current and new projects have a default environment setup?

Reply:
All environments so far have environment defined as it's a mandatory argument. This change makes it optional, so the config file (defaults to `.shaman.yml`) can be optional as well (all options can be set via the CLI arguments).

I don't expect any problems with existing applications. An issue might arise if a new caller/consumer doesn't specify the environment and doesn't have a mandatory option specified in CLI args. Eg.
```
bundle exec shaman deploy -m "New build" -n "1.0.1" -x abc-123
```
I've deliberately omitted the "release_path" option (eg. `-f "test.zip"`) which would cause an error when the system tries to determine the "release_path". Since "release_path" isn't specified in the CLI args, the system tries to load the config file (if it doesn't exist, the command failed) and then tries to "fetch" the environment. Since the environment isn't specified in the CLI args, the system will use the default environment name (also named `default`), and if a key with the same name isn't present in the loaded YAML, an error will be printed `Environment default doesn't exist in .shaman.yml`